test/coordinator-test.js

```diff
@@ -0,0 +1,29 @@
+import { assert } from 'chai';
+import os from 'os';
+import sinon from 'sinon';
+import { getDefaultCPUs, getWorkerCount } from '../lib/coordinator';
+
+describe('coordinator', () => {
+  it('default method returns correct number of cpus', () => {
+    assert.equal(getDefaultCPUs(5), 4, 'getDefaultCPUs returns n - 1 CPUs');
+    assert.equal(getDefaultCPUs(), 1, 'getDefaultCPUs falls back to 1 CPU');
+  });
+
+  it('uses the correct number of cpus', () => {
+    const sandbox = sinon.sandbox.create();
+    sandbox.stub(os, 'cpus').returns([{}, {}, {}, {}, {}]);
+
+    assert.equal(getWorkerCount(), 4, 'getWorkerCount defaults to all available cpus minus one');
```

Comment:
I haven't mocked get `getDefaultCPUs` out for the sake of test readability, but lmk if you think it should be.